modules/snooker/src/main/ts/ephox/snooker/model/RunOperation.ts

```diff
@@ -169,5 +179,37 @@ const onCells = (warehouse: Warehouse, target: TargetSelection): Optional<Detail
   return cells.length > 0 ? Optional.some(cells) : Optional.none();
 };
 
-export { run, toDetailList, onCell, onCells, onPaste, onPasteByEditor, onMergable, onUnmergable };
+// Custom unlocked extractors
+
+const onUnlockedCell = (warehouse: Warehouse, target: TargetElement): Optional<Structs.DetailExt> =>
+  onCell(warehouse, target).filter((detail) => !detail.isLocked);
```

Comment:
This isn't going to be ideal for performance on large tables, since it'll require multiple iterations. If we abstract the logic out in `onCell` we could reduce this to just a single iteration both here and in `onUnlockedCells`, so that might be worth looking into?

Here's a rough, untested example of how we might be able to do that:
```
const extractCells = (warehouse: Warehouse, target: TargetSelection, predicate: (detail: DetailExt) => boolean): Optional<DetailExt[]> => {
  const details = Arr.map(target.selection, (cell) => {
    return TableLookup.cell(cell)
      .bind((lc) => findInWarehouse(warehouse, lc))
      .filter(predicate);
  });
  const cells = Optionals.cat(details);
  return cells.length > 0 ? Optional.some(cells) : Optional.none();
};

Reply:
Good point and thanks for suggestion. I have made changes to the file with that in mind. Hopefully it is what you were thinking.

Reply:
Looks good thanks!